src/common/src/types/decimal.rs

```diff
@@ -499,6 +500,71 @@ impl Decimal {
         }
     }
 
+    #[must_use]
+    pub fn sin(&self) -> Self {
```

Comment:
FYI In PostgreSQL, trigonometric functions are only defined for double precision. Inputs of other types are [implicitly casted](https://github.com/risingwavelabs/risingwave/blob/c2decffad789a1e9361ab6c08634c8cc6aef603d/src/frontend/src/expr/type_inference/func.rs#L51) to double precision, and the output is always double precision.

Reply:
Thank you for pointing that out :)